vividus-plugin-rest-api/src/main/java/org/vividus/archive/steps/ArchiveSteps.java

```diff
@@ -0,0 +1,127 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.archive.steps;
+
+import static org.hamcrest.Matchers.hasItem;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jbehave.core.annotations.Then;
+import org.jbehave.core.annotations.When;
+import org.vividus.context.VariableContext;
+import org.vividus.model.ArchiveVariable;
+import org.vividus.model.NamedEntry;
+import org.vividus.softassert.ISoftAssert;
+import org.vividus.steps.DataWrapper;
+import org.vividus.steps.StringComparisonRule;
+import org.vividus.util.zip.ZipUtils;
+
+public class ArchiveSteps
+{
+    private final ISoftAssert softAssert;
+    private final VariableContext variableContext;
+
+    public ArchiveSteps(ISoftAssert softAssert, VariableContext variableContext)
+    {
+        this.softAssert = softAssert;
+        this.variableContext = variableContext;
+    }
+
+    /**
+     * Saves content of a file from archive to context variable in specified format
+     * Example:
+     * <p>
+     * <code>
+     * When I save content of `${response-as-bytes}` archive entries to variables:<br>
+     * |path                        |variableName|scopes   |outputFormat|
+     * |files/2011-11-11/skyrim.json|entityJson  |SCENARIO |TEXT        |
+     * </code>
+     * </p>
+     * Currently, available types are TEXT and BASE64
+     *
+     * @param archiveData The data of archive to verify
+     * @param parameters  describes saving parameters
+     */
+    @When("I save content of `$archiveData` archive entries to variables:$parameters")
+    public void saveFilesContentToVariables(DataWrapper archiveData, List<ArchiveVariable> parameters)
+    {
+        List<String> expectedEntries = parameters.stream().map(ArchiveVariable::getPath).collect(Collectors.toList());
+        Map<String, byte[]> zipEntries = ZipUtils.readZipEntriesFromBytes(archiveData.getBytes(),
+                expectedEntries::contains);
+        parameters.forEach(arcVar ->
+        {
+            String path = arcVar.getPath();
+            Optional.ofNullable(zipEntries.get(path)).ifPresentOrElse(
+                    data -> variableContext.putVariable(arcVar.getScopes(), arcVar.getVariableName(),
+                            arcVar.getOutputFormat().convert(data)),
+                    () -> softAssert.recordFailedAssertion(
+                            String.format("Unable to find entry by name %s in response archive", path)));
```

Comment:
The assertion message requires updates, because not every archive may be sourced from HTTP response. But at the same time we need to keep backward compatibility in case if someone uses these messages in known issues. So:
- this step should be updated to use relevant assertion message
- old step should be preserved and deprecated